MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -0,0 +1,116 @@
+     */
+    init(accessToken: String?, host: String?, tilesPath: String, translationsPath: String, completionHandler: @escaping OfflineDirectionsCompletionHandler)
+    
+    /**
+     Begins asynchronously calculating the route or routes using the given options and delivers the results to a closure.
+     
+     This method retrieves the routes asynchronously via MapboxNavigationNative.
+     
+     Routes may be displayed atop a [Mapbox map](https://www.mapbox.com/maps/). They may be cached but may not be stored permanently. To use the results in other contexts or store them permanently, [upgrade to a Mapbox enterprise plan](https://www.mapbox.com/directions/#pricing).
+     
+     - parameter options: A `RouteOptions` object specifying the requirements for the resulting routes.
+     - parameter completionHandler: The closure (block) to call with the resulting routes. This closure is executed on the application’s main thread.
+     */
+    func calculateOffline(_ options: RouteOptions, completionHandler: @escaping Directions.RouteCompletionHandler)
+}
+
+@objc(MBMapboxOfflineDirections)
+public class MapboxOfflineDirections: Directions, OfflineDirectionsProtocol {
+    
+    required public init(accessToken: String?, host: String?, tilesPath: String, translationsPath: String, completionHandler: @escaping OfflineDirectionsCompletionHandler) {
+        
+        super.init(accessToken: accessToken, host: host)
+        
+        OfflineDirectionsConstants.serialQueue.sync {
+            let tilesPath = tilesPath.replacingOccurrences(of: "file://", with: "")
+            let translationsPath = translationsPath.replacingOccurrences(of: "file://", with: "")
+            let tileCount = self.navigator.setupRouter(tilesPath, translationsPath: translationsPath)
+            
+            DispatchQueue.main.async {
+                completionHandler(tileCount)
+            }
+        }
+    }
+    
+    public func calculateOffline(_ options: RouteOptions, completionHandler: @escaping Directions.RouteCompletionHandler) {
+        
+        let url = self.url(forCalculating: options)
+        
+        OfflineDirectionsConstants.serialQueue.sync { [weak self] in
+            
+            guard let result = self?.navigator.getRouteForDirectionsUri(url.absoluteString) else {
+                return completionHandler(nil, nil, OfflineRoutingError.unexpectedRouteResult as NSError)
+            }
+            
+            guard let data = result.json.data(using: .utf8) else {
+                return completionHandler(nil, nil, OfflineRoutingError.corruptRouteData as NSError)
+            }
+            
+            do {
+                let json = try JSONSerialization.jsonObject(with: data, options: []) as! [String: Any]
+                if let errorValue = json["error"] as? String {
+                    DispatchQueue.main.async {
+                        let error = NSError(domain: "..", code: 102, userInfo: [NSLocalizedDescriptionKey: errorValue])
```

Comment:
This is an unusual domain. Can it be replaced with a constant that reliably identifies this SDK?